File: framework/languages/java/e2e/RegistrationCodec/Client/src/main/java/systems/zlink/e2e/registrationcodec/client/Scenarios/RcB1JsonCodecScenario.java
package systems.zlink.e2e.registrationcodec.client.Scenarios;

import java.time.Duration;
import systems.zlink.e2e.registrationcodec.client.Support.ScenarioAssert;
import systems.zlink.e2e.registrationcodec.client.Support.ScenarioContext;
import systems.zlink.e2e.registrationcodec.shared.Contracts;

public final class RcB1JsonCodecScenario {
    private static final Duration REQUEST_TIMEOUT = Duration.ofSeconds(5);

    private RcB1JsonCodecScenario() {
    }

    public static void run(ScenarioContext context) {
        Contracts.EchoReply jsonReply = context.client().requestToChannel(
                Contracts.CHANNEL,
                new Contracts.JsonEchoRequest("json-request"))
            .packetName("JsonEcho")
            .timeout(REQUEST_TIMEOUT)
            .await(Contracts.EchoReply.class);
        ScenarioAssert.ensure("echo:json-request".equals(jsonReply.value()) && "json".equals(jsonReply.handler()),
            "RC-B1 request mismatch");
        context.client().sendToChannel(Contracts.CHANNEL, new Contracts.JsonEchoCommand("json-send"))
            .packetName("JsonEcho")
            .await();
        ScenarioAssert.waitForEvidence(context.evidence(), "Send", "JsonEcho", "json-send");
        System.out.println("scenario RC-B1 passed");
    }
}

File: framework/languages/java/e2e-kotlin/RegistrationCodec/Client/src/main/kotlin/systems/zlink/e2e/kotlin/registrationcodec/client/Scenarios/RcB1JsonCodecScenario.kt
package systems.zlink.e2e.kotlin.registrationcodec.client.scenarios

import systems.zlink.e2e.kotlin.registrationcodec.Contracts
import systems.zlink.e2e.kotlin.registrationcodec.EchoReply
import systems.zlink.e2e.kotlin.registrationcodec.JsonEchoCommand
import systems.zlink.e2e.kotlin.registrationcodec.JsonEchoRequest
import systems.zlink.e2e.kotlin.registrationcodec.client.support.CodecScenarioResult
import systems.zlink.e2e.kotlin.registrationcodec.client.support.EvidenceText
import systems.zlink.e2e.kotlin.registrationcodec.client.support.ScenarioAssert
import systems.zlink.framework.channels.ZLinkClient

class RcB1JsonCodecScenario(
    private val client: ZLinkClient,
    private val evidence: EvidenceText,
    private val assert: ScenarioAssert,
) {
    fun run(): CodecScenarioResult {
        val jsonReply = client.requestToChannel(Contracts.CHANNEL, JsonEchoRequest("json-request"))
            .packetName("JsonEcho")
            .timeout(requestTimeout)
            .await(EchoReply::class.java)
        assert.that(jsonReply.value == "echo:json-request" && jsonReply.handler == "json", "RC-B1 request mismatch")
        client.sendToChannel(Contracts.CHANNEL, JsonEchoCommand("json-send"))
            .packetName("JsonEcho")
            .await()
        evidence.waitForEvidence("Send", "JsonEcho", "json-send")
        println("scenario RC-B1 passed")
        return CodecScenarioResult("RC-B1", "JsonEcho", jsonReply.value)
    }
}
